File: Month-1/Week-1/day-06/Go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Println("Input the number:")

	input, err := reader.ReadString('\n')

	if err != nil {
		fmt.Printf("Something went wrong: %v \n", err)
	} else {
		primeInput, _ := strconv.Atoi(strings.TrimSpace(input))
		result, _ := NextPrime(primeInput)

		fmt.Printf("The next prime number is: %v", result)
	}
}

func NextPrime(input int) (result int, err error) {

	if input <= 2 {
		return 2, err
	} else {
		for i := 2; i <= input; i++ {
			if input%i == 0 && i != input {
				return NextPrime(input + 1)
			} else {
				if i == input {
					return input, err
				}
			}
		}
	}

	return result, err
}

File: Month-1/Week-1/day-06/py/main.py

def is_prime(number):
    if (number < 2):
        return False
    i = 2
    while i < number:
        if number % i == 0:
            return False
        i += 1
    return True


def next_prime(number):
    number = int(number)
    return number if is_prime(number) else next_prime(number + 1)
